packages/lending/src/protocols/compound-v3/lending-protocol.ts

```diff
@@ -449,107 +285,55 @@ export class LendingProtocol extends Protocol {
 
     const borrows: BorrowObject[] = [
       {
-        token: unwrapToken(this.chainId, baseToken),
+        token: baseToken,
         price: baseTokenPrice,
         balances: [borrowBalance],
         apys: [borrowAPR],
       },
     ];
 
-    const portfolio = new Portfolio(this.chainId, this.id, marketId, supplies, borrows, baseToken);
+    const portfolio = new Portfolio(this.chainId, this.id, marketId, supplies, borrows);
 
     return portfolio;
   }
 
   async getPortfolios(account: string) {
-    return await Promise.all(
-      Object.keys(marketMap[this.chainId]).map((marketId) => this.getPortfolio(account, marketId))
-    );
+    return Promise.all(configMap[this.chainId].markets.map(({ id }) => this.getPortfolio(account, id)));
   }
 
-  async newSupplyLogic(params: SupplyParams) {
-    const { cometAddress, baseToken } = getMarketBaseConfig(this.chainId, params.marketId);
-    const cToken = await this.getToken(cometAddress);
-
-    const userBalances = await this.getUserBalances(params.marketId, params.account);
+  async newSupplyLogic({ marketId, input }: TokenInFields) {
+    const { baseToken, cToken } = getMarketConfig(this.chainId, marketId);
 
-    if (params.input.token.unwrapped.is(baseToken.unwrapped)) {
-      if (new BigNumberJS(userBalances.borrowBalance)) {
-        throw new Error('borrow USD is not zero');
-      }
-      const supplyQuotation = await apisdk.protocols.compoundv3.getSupplyBaseQuotation(this.chainId, {
-        input: params.input,
-        tokenOut: cToken,
-        marketId: params.marketId,
+    if (input.token.wrapped.is(baseToken)) {
+      return apisdk.protocols.compoundv3.newSupplyBaseLogic({
+        marketId,
+        input,
+        output: new common.TokenAmount(cToken, input.amount),
       });
-      return apisdk.protocols.compoundv3.newSupplyBaseLogic({ ...supplyQuotation, balanceBps: common.BPS_BASE });
     } else {
-      return apisdk.protocols.compoundv3.newSupplyCollateralLogic({
-        input: params.input,
-        marketId: params.marketId,
-        balanceBps: common.BPS_BASE,
-      });
+      return apisdk.protocols.compoundv3.newSupplyCollateralLogic({ marketId, input });
     }
   }
 
-  async newWithdrawLogic(params: WithdrawParams) {
-    const { cometAddress, baseToken } = getMarketBaseConfig(this.chainId, params.marketId);
-    const cToken = await this.getToken(cometAddress);
-
-    const userBalances = await this.getUserBalances(params.marketId, params.account);
+  async newWithdrawLogic({ marketId, output }: TokenOutFields) {
+    const { baseToken, cToken } = getMarketConfig(this.chainId, marketId);
 
-    const realAmount = new common.TokenAmount(params.output.token, params.output.amount);
-    realAmount.subWei(2);
```

Comment:
wondering why we removed this. This is related to 1 wei issue and will cause compound v3 withdraw fail

Reply:
@chouandy is currently updating the encoding methods for each operation in `adapter.ts`. Issues related to `BalanceLink` and `wei` should ideally be handled in `adapter.ts`, rather than in individual lending-protocol files.